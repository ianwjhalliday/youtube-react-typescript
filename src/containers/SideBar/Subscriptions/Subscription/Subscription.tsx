import React from 'react';
import {Icon, Image, Menu} from 'semantic-ui-react';
import './Subscription.scss';

export interface SubscriptionProps {
  broadcastingNow?: boolean;
  label: string;
  newVideosCount?: number;
}

export function Subscription(props: SubscriptionProps) {
  const {broadcastingNow, label, newVideosCount} = props;
  const rightElement = broadcastingNow
    ? <Icon name="signal" />
    : newVideosCount
      ? <span className="new-videos-count">{newVideosCount}</span>
      : null;

  return (
    <Menu.Item className="sidebar-item">
      <div className="subscription">
        <div>
          <Image src="http://via.placeholder.com/28x28" avatar />
          <span>{label}</span>
        </div>
        {rightElement}
      </div>
    </Menu.Item>
  );
}

export default Subscription;
