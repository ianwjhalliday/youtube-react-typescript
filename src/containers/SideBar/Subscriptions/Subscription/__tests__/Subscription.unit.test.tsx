import React from 'react';
import {shallow} from 'enzyme';
import Subscription from '../Subscription';

describe('renders Subscription', () => {
  test('with label', () => {
    const wrapper = shallow(
      <Subscription label="Productioncoder" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('with label and broadcastingNow', () => {
    const wrapper = shallow(
      <Subscription broadcastingNow label="Productioncoder" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('with label and newVideosCount', () => {
    const wrapper = shallow(
      <Subscription label="Productioncoder" newVideosCount={20} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('with label and broadcastingNow takes priority over newVideosCount', () => {
    const wrapper = shallow(
      <Subscription broadcastingNow label="Productioncoder" newVideosCount={20} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
