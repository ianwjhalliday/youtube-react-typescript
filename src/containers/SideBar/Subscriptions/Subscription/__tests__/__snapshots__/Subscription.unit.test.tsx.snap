// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Subscription with label 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Productioncoder
      </span>
    </div>
  </div>
</MenuItem>
`;

exports[`renders Subscription with label and broadcastingNow 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Productioncoder
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`renders Subscription with label and broadcastingNow takes priority over newVideosCount 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Productioncoder
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`renders Subscription with label and newVideosCount 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Productioncoder
      </span>
    </div>
    <span
      className="new-videos-count"
    >
      20
    </span>
  </div>
</MenuItem>
`;
